### PR TITLE
reset module context after invoking callback + use context manager (#990)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #990

### CODE REVIEW

File: src/taipy/gui/gui.py [hunk 1]
```diff
@@ -1138,7 +1138,10 @@ def _call_user_callback(
                 self.__set_client_id_in_context(state_id)
                 if module_context is not None:
                     self._set_locals_context(module_context)
-                return self._call_function_with_state(user_callback, args)
+                callback_result = self._call_function_with_state(user_callback, args)
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py]: Wasn't a 'finally' statement ever more clear?

**Review by FabienLelaquais — APPROVED** [review]
Fine with me

File: src/taipy/gui/gui.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply src/taipy/gui/gui.py]: and it would allow for exception management

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
definitely "finally"

**Review by FabienLelaquais — APPROVED** [review]
Indeed faaar more readable.
Well done!

**Review by FabienLelaquais — APPROVED** [review]
Nice!

**Pull request merged**